### PR TITLE
Recode with table

### PR DESCRIPTION
**Pull request by DarkShadeKnigh**
More or less final version of generic RecWTable. @DougManuel  and @wyusuf068  Check if the functionality is satisfactory.

### CODE REVIEW

File: R/strings.R [hunk 1]
```diff
@@ -6,15 +6,27 @@ pkg.globals$columnNames.Outlier <- "outlier"
...
 pkg.globals$argument.VariableDetailsSheet <- "variableDetailsSheet" 
 pkg.globals$argument.VariableStart <- "variableStart" 
 pkg.globals$argument.VariableStartType <- "variableStartType"
+pkg.globals$argument.DatabaseStart <- "databaseStart"
 pkg.globals$argument.VariableStartHigh <- "high" 
 pkg.globals$argument.VariableStartLow <- "low" 
 pkg.globals$argument.CatStartValue <- "value"
 pkg.globals$argument.CatStartLabel <- "valueLabelStart"
 pkg.globals$argument.VariableStartLabel <- "label"
+pkg.globals$argument.From <- "RecFrom"
```
Comment (DougManuel) [inline R/strings.R]: Let's make all these camelCase (as opposed to PascalCase). `recFrom` instead of `RecFrom`, etc.

File: R/strings.R [hunk 2]
```diff
@@ -6,15 +6,27 @@ pkg.globals$columnNames.Outlier <- "outlier"
...
+pkg.globals$argument.Interval <- "interval"
+pkg.globals$argument.CatValue <- "RecTo"
```
Comment (DougManuel) [inline R/strings.R]: recTo

File: R/strings.R [hunk 3]
```diff
@@ -6,15 +6,27 @@ pkg.globals$columnNames.Outlier <- "outlier"
...
+pkg.globals$argument.Notes <- "Notes"
```
Comment (DougManuel) [inline R/strings.R]: notes

File: R/strings.R [hunk 4]
```diff
@@ -6,15 +6,27 @@ pkg.globals$columnNames.Outlier <- "outlier"
...
+pkg.globals$argument.ToType <- "ToType"
```
Comment (DougManuel) [inline R/strings.R]: toType

File: inst/extdata/Copy of PBC Model Specification Workbook - variableDetails (1).csv [hunk 5]
```diff
@@ -0,0 +1,17 @@
+variable,variableType,variableDummy,valueLabelLong,valueLabel,target,method,reference,validCatN,catValues,units,low,high,interval,From,databaseStart,variableStart,variableStartType,catStartValue,catStartLabel,variableStartLow,variableStartHigh,variableStartLabel
```
Comment (DougManuel) [inline inst/extdata/Copy of PBC Model Specification Workbook - variableDetails (1).csv]: Could we have only one `variableDetails.csv` file? 
Suggest the best name is `PBC-variableDetails.csv`

Comment (DougManuel) [reply inst/extdata/Copy of PBC Model Specification Workbook - variableDetails (1).csv]: For `bllflow` use only the PBC data. (so, delete PORT MSW from this package and have only in chhsflow

File: inst/extdata/PoRT MSW - (Warsame)variableDetails.csv [hunk 6]
```diff
@@ -0,0 +1,996 @@
+Lime question,-----current variable name,variable,dummyVariable,variableType,method,target,databaseStart,variableStart,variableStartType,catValues,numValidCat,catLabel,catLabelLong,units,low,high,From,interval,catStartLabel,reference,variableStartShortLabel,variableStartLabel,Notes
```
Comment (DougManuel) [inline inst/extdata/PoRT MSW - (Warsame)variableDetails.csv]: delete this file from bllflow. Have only in cchsflow

**Comment by DougManuel** [comment]
I added the description to the recode-with-table.R file. @wyusuf068 can you try finishing?

File: R/recode-with-table.R [hunk 7]
```diff
@@ -116,9 +120,35 @@ RecWTable.default <-
     
     return(dataSource)
   }
-#' Get Data Variable Name
-#'
-#' Retrieves the name of the column inside dataSource to use for calculations
+
+#' @details The \code{variableDetails} dataframe has the following variables:
```
Comment (DarkShadeKnigh) [inline R/recode-with-table.R]: Why is this attached on the GetDataVariableName function? Wouldnt this be better on the RecWTable function?

Comment (DarkShadeKnigh) [reply R/recode-with-table.R]: @DougManuel

**Comment by DarkShadeKnigh** [comment]
Since cchsflow is going public we need this pushed to master

**Comment by DougManuel** [comment]
@yulric Do you want to give a quick review -- or is it OK to close after checks pass.

**Comment by DarkShadeKnigh** [comment]
@yulric  Ready for review

**Pull request merged**